wp/lib/bap_wp/src/precondition.ml

```diff
@@ -594,14 +632,15 @@ let visit_jmp (env : Env.t) (post : Constr.t) (jmp : Jmp.t) : Constr.t * Env.t =
                            The loop handler should have added the precondition for the node");
             in
             let cond = Jmp.cond jmp in
-            let cond_val, assume, vcs, env = exp_to_z3 cond env in
-            debug "\n\nJump when %s:\nVCs:%s\nAssumptions:%s\n\n%!"
-              (Expr.to_string cond_val) (List.to_string ~f:Constr.to_string vcs)
-              (List.to_string ~f:Constr.to_string assume);
+            let cond_val, hooks, env = exp_to_z3 cond env in
+            debug "\n\nJump when %s:\n%s\n%!"
+              (Expr.to_string cond_val) (hooks_to_string hooks);
             let cond_size = BV.get_size (Expr.get_sort cond_val) in
             let false_cond = Bool.mk_eq ctx cond_val (z3_expr_zero ctx cond_size) in
             let ite = Constr.mk_ite jmp (Bool.mk_not ctx false_cond) l_pre post in
-            let post = ite::vcs in
+            let vcs = hooks.verify_before @ hooks.verify_after in
```

Comment:
Maybe we should be a little careful here: if we add a `PC` variable, these won't be identical anymore...

Reply:
Not sure what the fix is, other than adding a comment...